pkg/operator/status.go

```diff
@@ -148,22 +148,33 @@ func (optr *Operator) syncDegradedStatus(ierr error) (err error) {
 	optrVersion, _ := optr.vStore.Get("operator")
 	degraded := configv1.ConditionTrue
 	var message, reason string
-	if ierr == nil {
+	if ierr.error == nil {
 		degraded = configv1.ConditionFalse
+		reason = ""
 	} else {
 		if optr.vStore.Equal(co.Status.Versions) {
 			// syncing the state to exiting version.
 			message = fmt.Sprintf("Failed to resync %s because: %v", optrVersion, ierr.Error())
 		} else {
 			message = fmt.Sprintf("Unable to apply %s: %v", optrVersion, ierr.Error())
 		}
-		reason = failedToSyncReason
+		reason = ierr.task + "SyncError"
```

Comment:
ok sorry I said `SyncError` but I think `Failed` would be better and more aligned with other telemtry coming from other opererators (sorry for the back and forth)